Bottom: d650821e2d23132d1ac65651fe2e76598d562188
Top:    cd48268efc94b52ce52e38716814999c1eeab0a8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 20:31:13 +0000

Catch incomplete step data without crashing

XXX: May be plenty of other cases where we assume certain data is present!


---

diff --git a/src/hid/step/utils.cpp b/src/hid/step/utils.cpp
index ea90a52..e44b78a 100644
--- a/src/hid/step/utils.cpp
+++ b/src/hid/step/utils.cpp
@@ -70,7 +70,15 @@ find_all_pd_with_sdr (InstMgr *instance_list, pd_list *pd_list)
     {
       SdaiRepresented_definition *sdr_definition = sdr->definition_ ();
       SdaiProduct_definition_shape *pds = (SdaiProduct_definition_shape *)(SdaiProperty_definition_ptr)(*sdr_definition);
-      SdaiProduct_definition *pd = *(SdaiCharacterized_product_definition_ptr)(*pds->definition_ ());
+      SdaiProduct_definition *pd;
+
+      if (pds == NULL)
+        {
+          fprintf (stderr, "pds == NULL in find_all_pd_with_sdr\n");
+          return;
+        }
+
+      pd = *(SdaiCharacterized_product_definition_ptr)(*pds->definition_ ());
 
       pd_list->push_back (pd);
